modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/NodeImpl.java

```diff
@@ -1645,6 +1640,23 @@ private void executeApplyingTasks(final List<LogEntryAndClosure> tasks) {
         }
     }
 
+    /**
+     * Builds a status for 'Cannot apply because this node is not a leader' situation.
+     *
+     * @param nodeState Node state.
+     */
+    public static Status cannotApplyBecauseNotLeaderStatus(State nodeState) {
```

Comment:
There's nothing about "cannot apply" in the status itself. Why did you chose such a specific name?

Reply:
Because this method checks for this specific situation. When it finds out that 'it cannot apply because not a leader', it returns this status.

Reply:
From what I see, method knows nothing about "cannot apply", it only converts one status to another, that's why I asked the question. But I see what you mean